src/btf.cpp

```diff
@@ -1072,7 +1073,7 @@ ast::Pass CreateParseBTFPass()
 {
   return ast::Pass::create(
       "btf", []([[maybe_unused]] ast::ASTContext &ast, BPFtrace &b) {
-        b.parse_btf(b.list_modules(ast));
+        b.parse_module_btf(b.list_modules(ast));
```

Comment:
Correct me if I'm wrong but with this change, this pass no longer parses the entire BTF, just the modules BTF. The vmlinux BTF is parsed in `BTF` constructor which must be called before this pass is executed (otherwise `btf_` in `BPFtrace::parse_module_btf` is null).

Is that the expected behaviour? Are we sure that it's safe to do?

Reply:
> I don't think so, it was created by

Ah you're right. I should have actually looked instead of trying to recall from memory 🧠 

> So I guess that the only real change here is that vmlinux BTF is always parsed as opposed to the previous implementation

Actually it was always parsed in the previous implementation because a BTF object was constructed (and vmlinux parsed) by BPFfeature when constructing the BPFtrace object. Very tricky! But I agree with your point that we can possibly stop this in tests that don't need it. Let me look into that in a follow-up commit/PR.

Reply:
Absolutely, thanks!